### ISSUE TITLE
Undefined variable b:NERDTree

### ISSUE THREAD

**Issue by kekscode** [PhilRunninger/nerdtree-visual-selection#3]
First of all, all ~the functionality seem to work~ (thank you, great plugin!)
Update: It does not work, see [this comment](https://github.com/PhilRunninger/nerdtree-visual-selection/issues/3#issuecomment-515651094).

One issue though:

1. NERDTree is not visible
2. Type `:e .` to bring up nerd tree (this is maybe what i am doing "wrong" here)
3. Visually select a bunch of files in the nerd tree buffer and press `o` (NERDTreeMapActivateNode).

Result:

~All files are opened (functionality OK) but~ This error trace appears:

```
NERDTree: Opening /Users/san/devops/benthos/LICENSE (1 of 2)...                                                                                                        
Error detected while processing function <SNR>108_ProcessSelection:                                                                                                    
line    6:                                                                                                                                                             
E716: Key not present in Dictionary: path.str() . " (" . (curLine - a:firstline + 1) . " of " . (a:lastline - a:firstline + 1) . ")...")                               
E116: Invalid arguments for function nerdtree#echo                                                                                                                     
Error detected while processing function <SNR>108_ProcessSelection[16]..NERDTreeRender[1]..nerdtree#renderView:                                                        
line    1:                                                                                                                                                             
E121: Undefined variable: b:NERDTree
```

I don't get that error when using any other opening method except `NERDTreeMapActivateNode` and i don't get the error when the nerd tree is activated with `:NERDTree` before doing the above.

Can this issue be resolved?

**Comment by PhilRunninger** [comment]
There are two different kinds of NERDTrees; Marty calls them **TabTree** and **WindowTree**. The **TabTree** is the one that works. I'll have to dig into the NERDTree code to see what's going on here, but it looks like it doesn't have the same variables initialized when it's created. In the meantime, you can use `:NERDTree .` instead to create a **TabTree**.

**Comment by kekscode** [comment]
Thanks for the explanation and the suggested workaround. Appreciated!

**Comment by PhilRunninger** [comment]
@suchkultur , If you look again, ignoring the error messages, you'll see that it's not working right. While looping, the first file replaces the NERDTree, then all the rest of the files can't be opened because the NERDTree isn't there anymore. I will see if there is a way to work around this behavior, or display a message saying the function is unavailable.

**Comment by kekscode** [comment]
Oh i see. Sorry about my misleading comment at the beginning of this issue. I didn't take my time to check the behaviour properly, because i was actually in the middle of doing something completely different when i noticed the issue. I'll update the description.

**Comment by PhilRunninger** [comment]
Closed with #4.
